client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx

```diff
@@ -9,50 +9,72 @@ import Button from '@material-ui/core/Button';
 import Box from '@material-ui/core/Box';
 import DoneIcon from '@material-ui/icons/Done';
 
-import { useAuth } from '../../../context/useAuthContext';
-import { Contest } from '../../../interface/Contest';
 import useStyles from './useStyles';
 import SubmissionCardProps from '../SumissionCardInterface';
-import { fetchContestById } from '../../../helpers/APICalls/searchContest';
+import { Contest } from '../../../interface/Contest';
+import { User } from '../../../interface/User';
+import { Winner } from '../../../interface/Winner';
+import { postWinner } from '../../../helpers/APICalls/winners';
+import chargeCustomer from '../../../helpers/APICalls/chargeCustomer';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import deadlinePassed from '../../../helpers/datePassed';
 
 export interface ParamProps {
   id: string;
 }
 
-function SubmissionCard({ imageSrc, author }: SubmissionCardProps): JSX.Element {
+function SubmissionCard({ imageSrc, author, contest }: SubmissionCardProps): JSX.Element {
   const classes = useStyles();
   const { loggedInUser } = useAuth();
-  const [contest, setContest] = useState<Contest>();
-  const { id } = useParams<ParamProps>();
-
-  useEffect(() => {
-    const ac = new AbortController();
-    fetchContestById({ id: id }).then((res) => {
-      if (res.success) {
-        setContest(res.success as Contest);
-      }
-    });
-    return ac.abort();
-  }, [id]);
+  const { updateSnackBarMessage } = useSnackBar();
 
   if (loggedInUser === undefined || !loggedInUser || contest == undefined || contest.userID == undefined)
     return <CircularProgress />;
 
+  const handleWinnerSelection = async (
+    imageSrc: string,
+    author: User,
+    contest: Contest,
+    user: User | null | undefined,
+  ) => {
+    const winner: Winner = {
+      contestTitle: contest.title,
+      image: imageSrc,
+      username: author.username,
+      winningDate: new Date(),
+    };
+
+    const saveWinner = await postWinner(winner, contest._id);
+
+    if (saveWinner && saveWinner.winner) {
+      const processingPayment = await chargeCustomer(contest._id);
+
+      if (processingPayment && processingPayment.status) {
+        updateSnackBarMessage('Card on file has been successfully charged. Thank you');
+        // Send Email to the user and the winner
+        console.log(`Email will be sent to ${author.username} and ${user?.username}`);
+      }
+    }
+  };
+
   return (
     <Card className={classes.submissionCard}>
       <CardMedia component="img" alt="Contemplative Reptile" height="300" image={imageSrc} />
       <Box className={classes.displayOver}>
         <CardContent className={classes.submissionContent}>
           <Typography gutterBottom variant="h5" component="h2">
-            {loggedInUser.id == contest?.userID._id && (
-              <Button className={classes.winner}>
+            {deadlinePassed(contest.deadlineDate) ? (
+              <Button
```

Comment:
Is the Winner button currently visible to all users?

Reply:
Winner button will only be visible to the contest creator on his dashboard under completed contest, once the deadline is passed.